controllers/storageclient_controller.go

```diff
@@ -210,9 +217,25 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 			if err := r.reconcileClusterResourceQuota(clusterResourceQuotaSpec); err != nil {
 				return reconcile.Result{}, err
 			}
+		case "Secret":
+			if eResource.Name == fmt.Sprintf("noobaa-remote-join-secret-%s", r.storageClient.Status.ConsumerID) {
+				data := map[string]string{}
+				if err := json.Unmarshal(eResource.Data, &data); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to unmarshall noobaa join secret data: %v", err)
+				}
+
+				if err := r.reconcileNoobaaJoinSecret(data); err != nil {
+					return reconcile.Result{}, err
+				}
+			}
+		}
+	}
+	// reconcile  remote noobaa only if it is not a local client
+	if _, exists := r.storageClient.GetAnnotations()["is-local-client"]; !exists {
```

Comment:
could you pls explain why we need to differentiate b/n native vs remote client?

Reply:
We don't want to create a client noobaa on the same cluster where the noobaa system is present to which all other remote clients will connect to, hence the annotation to differentiate
Ref: https://github.com/red-hat-storage/ocs-operator/pull/2680/files#diff-3a3043bcbec115677bf1cd899c5c562a02eb74b2c874f80b42ff8bd6880c41f4R43

Reply:
No it won't. The nooba on the provider side is the one which consists of the main noobaa system, similar to what we have today, no change in existing functionality to which other clients would connect to